From a3831f4f5673c32a18d6197fa46e1d0af8cb2af4 Mon Sep 17 00:00:00 2001
From: johnmart19 <johnivan19999@gmail.com>
Date: Tue, 27 Sep 2022 18:38:14 +0300
Subject: [PATCH] Revert "kernel: Drop -fuse-ld=lld from CFLAGS"

This commit entirely breaks kernel compilation on 4.14 kernels

clang-14: error: unable to execute command: Executable "ld" doesn't exist!
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [scripts/Makefile.host:102: scripts/basic/fixdep] Error 1
make[1]: *** [/-path_ommited-/kernel/xiaomi/vayu/Makefile:499: scripts_basic] Error 2

This reverts commit 12ecb9f359983b774416b8910e8c1c048ec98635.
---
 config/BoardConfigKernel.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/config/BoardConfigKernel.mk b/config/BoardConfigKernel.mk
index ab96aaf..666c341 100644
--- a/config/BoardConfigKernel.mk
+++ b/config/BoardConfigKernel.mk
@@ -140,9 +140,9 @@ ifeq ($(TARGET_KERNEL_CLANG_COMPILE),false)
 endif
 
 ifeq ($(HOST_OS),darwin)
-  KERNEL_MAKE_FLAGS += HOSTCFLAGS="-I$(BUILD_TOP)/external/elfutils/libelf -I/usr/local/opt/openssl/include" HOSTLDFLAGS="-L/usr/local/opt/openssl/lib -fuse-ld=lld"
+  KERNEL_MAKE_FLAGS += HOSTCFLAGS="-I$(BUILD_TOP)/external/elfutils/libelf -I/usr/local/opt/openssl/include -fuse-ld=lld" HOSTLDFLAGS="-L/usr/local/opt/openssl/lib -fuse-ld=lld"
 else
-  KERNEL_MAKE_FLAGS += CPATH="/usr/include:/usr/include/x86_64-linux-gnu" HOSTLDFLAGS="-L/usr/lib/x86_64-linux-gnu -L/usr/lib64 -fuse-ld=lld"
+  KERNEL_MAKE_FLAGS += CPATH="/usr/include:/usr/include/x86_64-linux-gnu" HOSTCFLAGS="-fuse-ld=lld" HOSTLDFLAGS="-L/usr/lib/x86_64-linux-gnu -L/usr/lib64 -fuse-ld=lld"
 endif
 
 TOOLS_PATH_OVERRIDE := \
-- 
2.37.3

